src/trustRegion.jl

```diff
@@ -246,6 +249,7 @@ function SciMLBase.__init(prob::NonlinearProblem{uType, iip}, alg::TrustRegion,
     else
         u = deepcopy(prob.u0)
     end
+    u_prev = deepcopy(u)
```

Comment:
This is only needed by some algorithms though, right? So in many cases it can be made as an empty array?